core/store/src/trie/shard_tries.rs

```diff
@@ -13,18 +13,20 @@ use near_primitives::types::{
 
 use crate::db::{DBCol, DBOp, DBTransaction};
 use crate::trie::trie_storage::{TrieCache, TrieCachingStorage};
-use crate::trie::TrieRefcountChange;
+use crate::trie::{TrieRefcountChange, POISONED_LOCK_ERR};
 use crate::{StorageError, Store, StoreUpdate, Trie, TrieChanges, TrieUpdate};
 
-#[derive(Clone)]
-pub struct ShardTries {
-    pub(crate) store: Arc<Store>,
+struct ShardTriesInner {
+    store: Arc<Store>,
```

Comment:
Do we need this Arc here?

Reply:
Going to be removed in https://github.com/near/nearcore/pull/4817#discussion_r708279480 I suppose.